webapp/tests/test_finders.py

```diff
@@ -153,6 +158,32 @@ def listdir_mock(d):
             os.listdir = self._original_listdir
             self.wipe_whisper()
 
+    def test_standard_finder_gzipped_whisper(self):
+        def listdir_mock(d):
+            self._listdir_counter += 1
+            return self._original_listdir(d)
+
```

Comment:
This is entirely reasonable, but stupid question. Why not just use `mock.patch()` with a context manager?